server/index.js

```diff
@@ -6,13 +6,23 @@ import db from './database/db.js';
 dotenv.config();
 const app = express();
 const PORT = process.env.PORT || 4000;
-db();
 
 app.use(express.json());
 app.use('/api/auth', routes);
 
-try {
-    app.listen(PORT, () => console.log(`Server started on port ${PORT}`));
-    } catch (error) {
-        console.error("connection failed");
-    }
\ No newline at end of file
+ async function startApp(){
```

Comment:
This is valid, but it's just redundant code, try this

```javascript
db()
   .then(() => {
       console.log("DB: running")
       app.listen(PORT, () => {
            console.log(`Server started on port ${PORT}`)
        });
   })
   .catch(console.log)
```

Reply:
This way, you'll handle the error, and also start the server only if DB connection is successful, also less code

Reply:
Okay